.github/workflows/publish.yml

```diff
@@ -32,11 +32,69 @@ jobs:
         run: echo "TAG=$(echo ${GITHUB_REF:10})" >> $GITHUB_ENV
 
       - name: Publishing app to Google Play
+        env:
+          KEY_ALIAS: ${{ secrets.KEY_ALIAS }}
+          KEY_PASSWORD: ${{ secrets.KEY_PASSWORD }}
+          KEY_STORE_PASSWORD: ${{ secrets.KEY_STORE_PASSWORD }}
+          DWDS_HTTP_BASIC_ACCESS_AUTHENTICATION: ${{ secrets.DWDS_HTTP_BASIC_ACCESS_AUTHENTICATION }}
```

Comment:
Should be generic name.

Reply:
@gouri-panda We only have the `DWDS` app which requires the authentication that's why we have done this as requested in https://github.com/kiwix/kiwix-android-custom/issues/107.

If you are thinking we should give a generic name for this to make this process for every app(if any app will require authentication), then we need to modify our URL according to the generic name because we are extracting secretKey from the URL.
```kotlin
fun ProductFlavor.fetchRequest(): Request {
  val urlString = buildConfigFields["ZIM_URL"]!!.value.replace("\"", "")
  return if (urlString.isAuthenticationUrl) {
    Request.Builder()
      .url(URI.create(urlString.removeAuthenticationFromUrl).toURL())
      .header(
        "Authorization",
        "Basic " +
          Base64.getEncoder().encodeToString(System.getenv(urlString.secretKey).toByteArray())
      )
      .build()
  } else {
    Request.Builder()
      .url(URI.create(urlString).toURL())
      .build()
  }
}
```

And for future apps we should have the same `authenticationValue` for every app. If you want to just make it generic and not thinking about future apps, than we just need to modify our URL, something like 

```
{
  "app_name": "Dwds",
  "zim_url": "https://{{BASIC_ACCESS_AUTHENTICATION}}@www.dwds.de/kiwix/f/dwds_de_dictionary_nopic_2023-09-12.zim",
  "enforced_lang": "de"
}
```

And we need to modify the secret in Github then we can do this.

```
KEY_ALIAS: ${{ secrets.KEY_ALIAS }}
KEY_PASSWORD: ${{ secrets.KEY_PASSWORD }}
KEY_STORE_PASSWORD: ${{ secrets.KEY_STORE_PASSWORD }}
BASIC_ACCESS_AUTHENTICATION: ${{ secrets.BASIC_ACCESS_AUTHENTICATION }}
```

@gouri-panda, @kelson42  Let me know what are your thoughts on this?

Reply:
dwds credentials can not have a generic name...